app/lib/actions/katello/repository/upload_package_group.rb

```diff
@@ -0,0 +1,34 @@
+module Actions
+  module Katello
+    module Repository
+      class UploadPackageGroup < Actions::EntryAction
+        def plan(repository, params)
+          action_subject(repository)
+          unit_key = {"repo_id": repository.pulp_id, "id": params[:name].parameterize.underscore}
+          metadata = {"unit_metadata": params}
```

Comment:
Also, you can't pass in all the params from the controller, they have to be filtered out (or else pulp will error).  i'd add:

```
          params = params.slice(:name, :description, :user_visible, :mandatory_package_names, :optional_package_names, :conditional_package_names, :default_package_names)

```